packages/datadog-plugin-prisma/test/index.spec.js

```diff
@@ -7,299 +7,433 @@ const path = require('node:path')
-          const matched = version.match(/(\d+)\.\d+\.\d+$/)
-          const majorVersion = matched[1]
-          tracingHelper = global.PRISMA_INSTRUMENTATION?.helper ||
-            global[`V${majorVersion}_PRISMA_INSTRUMENTATION`]?.helper
-        })
 
-        after(() => {
-          return agent.close({ ritmReset: false })
-        })
-
-        it('should do automatic instrumentation', async () => {
-          const tracingPromise = agent.assertSomeTraces(traces => {
-            assert.strictEqual(traces[0][0].resource, 'queryRaw')
-            assert.strictEqual(traces[0][0].meta['prisma.type'], 'client')
-            assert.strictEqual(traces[0][0].meta['prisma.method'], 'queryRaw')
-            assert.strictEqual(traces[0][0].name, expectedSchema.client.opName)
-            assert.strictEqual(traces[0][0].service, expectedSchema.client.serviceName)
-
-            // grabbing actual db query span
-            const engineDBSpan = traces[0].find(span => span.meta['prisma.name'] === 'db_query')
-            assert.strictEqual(engineDBSpan.resource, 'SELECT 1')
-            assert.strictEqual(engineDBSpan.type, 'sql')
-            assert.strictEqual(engineDBSpan.meta['span.kind'], 'client')
-            assert.strictEqual(engineDBSpan.name, expectedSchema.engine.opName)
-            assert.strictEqual(engineDBSpan.service, expectedSchema.engine.serviceName)
+          after(() => {
+            return agent.close({ ritmReset: false })
           })
 
-          await Promise.all([
-            prismaClient.$queryRaw`SELECT 1`,
-            tracingPromise
-          ])
-        })
-
-        it('should handle errors', async () => {
-          let error
-          const tracingPromise = agent.assertFirstTraceSpan((trace) => {
-            assertObjectContains(trace, {
-              meta: {
-                [ERROR_TYPE]: error.name,
-                [ERROR_MESSAGE]: error.message,
-                [ERROR_STACK]: error.stack,
+          it('should do automatic instrumentation', async () => {
+            const tracingPromise = agent.assertSomeTraces(traces => {
+              assert.strictEqual(traces[0][0].resource, 'queryRaw')
+              assert.strictEqual(traces[0][0].meta['prisma.type'], 'client')
+              assert.strictEqual(traces[0][0].meta['prisma.method'], 'queryRaw')
+              assert.strictEqual(traces[0][0].name, expectedSchema.client.opName)
+              assert.strictEqual(traces[0][0].service, expectedSchema.client.serviceName)
+
+              // grabbing actual db query span
+              if (config.v7) {
+                const pgSpan = traces[0].find(span => span.name === 'pg.query')
+                assert.strictEqual(pgSpan.resource, 'SELECT 1')
+                assert.strictEqual(pgSpan.type, 'sql')
+                assert.strictEqual(pgSpan.meta['span.kind'], 'client')
+                assert.strictEqual(pgSpan.name, 'pg.query')
+                assert.strictEqual(pgSpan.service, 'test-postgres')
```

Comment:
```suggestion
                assertObjectContains(
                  traces[0],
                  {
                    resource: 'SELECT 1',
                    type: 'sql',
                    meta: {
                      'span.kind': 'client'
                    },
                    name: 'pg.query',
                    service: 'test-postgres'
                  }
                )
```

Using this helper is favorable due to getting much better errors. A rule that I use: the less assertions for a single input, the better.